packages/chain-mon/src/wd-mon/service.ts

```diff
@@ -109,10 +109,18 @@ export class WithdrawalMonitor extends BaseServiceV2<Options, Metrics, State> {
     // Not detected by default.
     this.state.forgeryDetected = false
 
-    // For now we'll just start take it from the env or the tip of the chain
+    this.state.faultProofWindow =
+      await this.state.messenger.getChallengePeriodSeconds()
+    this.logger.info(
+      `fault proof window is ${this.state.faultProofWindow} seconds`
+    )
+
+    // Set the start block number.
     if (this.options.startBlockNumber === -1) {
+      // We default to starting from the earliest block still in the fault proof window.
+      const l1BlockNumber = await this.options.l1RpcProvider.getBlockNumber()
       this.state.highestUncheckedBlockNumber =
-        await this.options.l1RpcProvider.getBlockNumber()
+        l1BlockNumber - this.state.faultProofWindow / 12
```

Comment:
Just for sanity purposes, can we walk backwards from this estimated block height and terminate when reaching a block outside of the fpw? Would be great to ensure the right height

Reply:
OK, just to make sure I'm following that'd mean:
1. get the timestamp from the latest block. 
2. query for a block number based on the math here
3. check if that block has a timestamp from before the start of the FPW
4. If not, try the previous block, etc.